src/js/contexts/GlobalContext.js

```diff
@@ -0,0 +1,68 @@
+import React, { Component, createContext } from "react";
+import {iconClassNames, uiClassNames} from "../components/ServerButton/ServerButton";
+
+export const GlobalContext = createContext('');
+
+export class GlobalContextProvider extends Component {
+  state = {
+    currentMainNavButtonId: "home",
+    mainNavButtons: [
+      {
+        id: "home",
+        title: "Home",
+        iconClassName: iconClassNames.home,
+        channelExtraClassNames: [ uiClassNames.blue, uiClassNames.separator ],
+      },
+      {
+        id: "pythos-server",
+        title: "Pytho's Server",
+        imageSrc: null,
+        channelExtraClassNames: [ uiClassNames.blue ],
+      },
+      {
+        id: "add-a-server",
+        title: "Add a Server",
+        iconClassName: iconClassNames.add,
+        channelExtraClassNames: [ uiClassNames.green ],
+      },
+      {
+        id: "server-discovery",
+        title: "Server Discovery",
+        iconClassName: iconClassNames.discover,
+        channelExtraClassNames: [ uiClassNames.green, uiClassNames.separator ],
+      },
+      {
+        id: "download-apps",
+        title: "Download Apps",
+        iconClassName: iconClassNames.download,
+        channelExtraClassNames: [ uiClassNames.green ],
+      },
+    ],
+  }
+  /**
+   *  Sets 'currentMainNavButtonId' in state to give the active server button its' active styling
+   * @param id
+   */
+  setCurrentMainNavButtonId = (id) => this.setState({ currentMainNavButtonId: id });
+
+  addNewServerButton = (button) => {
+    const currentButtonsState = this.state.mainNavButtons
+
+    currentButtonsState.splice(-3, 0, button)
```

Comment:
Good enough for this build. But there is an assumption here that the insert point is always `-3`, so in a larger project it would be good to consider how to make eliminate that assumption.